packages/data-broker/src/tests/services/weather/DateSanitiser.test.js

```diff
@@ -0,0 +1,123 @@
+import { expect } from 'chai';
+import { DateSanitiser } from '../../../services/weather/DateSanitiser';
+import { mockNow, resetMocks } from './testutil';
+
+describe('DateSanitiser', () => {
+  beforeEach(() => {
+    mockNow(1549360800 * 1000); // (2019-02-05 10:00 UTC)
+  });
+
+  afterEach(() => {
+    resetMocks();
+  });
+
+  it('throws when start > end', () => {
+    const functionCall = () => new DateSanitiser().sanitise('2019-01-20', '2019-01-10');
+    expect(functionCall).to.throw('Start date must be before (or equal to) end date');
+  });
+
+  describe('limits on earliest data', () => {
+    /*
+     * This is a limitation of the WeatherBit plan we use, if we request further back it falls over
+     */
+
+    it('both start and end dates are before the earliest available data', () => {
+      const { startDate: actualStartDate, endDate: actualEndDate } = new DateSanitiser().sanitise(
+        '2017-02-10',
+        '2017-02-20',
+      );
+
+      expect(actualStartDate).to.equal('2018-02-05'); // earliest date
+      expect(actualEndDate).to.equal('2018-02-06'); // earliest + 1 day
+    });
+
+    it('only start date is before the earliest available data', () => {
+      const { startDate: actualStartDate, endDate: actualEndDate } = new DateSanitiser().sanitise(
+        '2017-02-10',
+        '2019-01-01',
+      );
+
+      expect(actualStartDate).to.equal('2018-02-05'); // earliest date
+      expect(actualEndDate).to.equal('2019-01-01');
+    });
+
+    it('only start date is before earliest, end date is already earliest', () => {
+      const { startDate: actualStartDate, endDate: actualEndDate } = new DateSanitiser().sanitise(
+        '2017-02-10',
+        '2018-02-06',
+      );
+
+      expect(actualStartDate).to.equal('2018-02-05'); // earliest date
+      expect(actualEndDate).to.equal('2018-02-06'); // (same as input) earliest + 1 day
+    });
+  });
+
+  describe('limits on latest data', () => {
+    /*
+     * Today's data is incomplete, e.g. we don't know the total rainfall for the day until
+     * the day is over. So we cannot return data for today, instead we return data for yesterday.
+     *
+     * There may be a possibility of using the forecast API to get today's forecast max temp / precip,
+     * but this will be covered by #1250
+     *
+     * Note for latest available dates:
+     *  - for WeatherBit API, end date today means at 00:00:01 today, which does not cover any weather data for today
```

Comment:
Right! This answers my earlier question. That's a helpful comment, thanks.

I don't think the date sanitiser should be "deceptive", if you know what I mean. Sure, it may constrain the endpoints so that they fall within the accepted range, but I don't think it should "shift" the date range that is being requested. I like that you've got a separate and clear code path for requesting yesterday's data in the service, when no date range is provided. However, if a client chooses to select a date range that explicitly requests today, I think we should just return no results, or throw an error.

Reply:
Discussed, using forecast api and different data elements for current day, so DateSanitiser no longer needs to be sneaky